### PR TITLE
update hvac by including poetry-core and pinning zipp

### PR DESCRIPTION
**Pull request by addyess**
Requires https://github.com/juju/charm-helpers/pull/745 first be merged and published to pypi before merging
* https://github.com/juju/charm-helpers/pull/745

The above is merged, but a new charmhelpers must be published to pypi

### CODE REVIEW

File: lib/charms/layer/vault_kv.py [hunk 1]
```diff
@@ -57,7 +57,9 @@ def _client(self):
                 vault_url=self._config["vault_url"],
             )
             client = hvac.Client(url=self._config["vault_url"])
-            client.auth_approle(self._config["role_id"], self._config["secret_id"])
+            client.auth.approle.login(
+                self._config["role_id"], self._config["secret_id"]
+            )
```
Comment (addyess) [inline lib/charms/layer/vault_kv.py:62]: just black

**Review by kwmonroe — APPROVED** [review]
Changes seem sound, but I'm not sure when we'll see an official pypi release for the requisite charmhelpers. Can we explicitly pin ch to a tag/commit until a release is available?

**Comment by addyess** [comment]
> Changes seem sound, but I'm not sure when we'll see an official pypi release for the requisite charmhelpers. Can we explicitly pin ch to a tag/commit until a release is available?

It feels like the pinning should be in the charm, not at the layer level... i dunno why though.

**Comment by addyess** [comment]
[Here's a pinning of charmhelpers](https://github.com/charmed-kubernetes/charm-kubernetes-control-plane/pull/266) which is the only i can find using this layer

**Review by kwmonroe — APPROVED** [review]

**Pull request merged**